gui/wxpython/startup/guiutils.py

```diff
@@ -692,6 +697,69 @@ def delete_grassdb_interactively(guiparent, grassdb):
     return deleted
 
 
+def switch_mapset_interactively(guiparent, grassdb, location, mapset):
```

Comment:
Let's rename this to `can_switch_mapset_interactively` and it won't be actually switching the mapset, just checking and removing the lockfile. The signals would be called from tree.py as before. This function could be also used in lmgr/frame.py in OnChangeMapset/Location.